### ISSUE TITLE
`pnpm update` removes package if there was a semver major update

### ISSUE THREAD

**Issue by anantoghosh** [pnpm/pnpm#1333]
package still exists in shrinkwrap and node_modules but is removed from package.json

### pnpm version:
2.13.5

### Code to reproduce the issue:
```
PS D:\Develop\pnpmtests> pnpm install chalk@1.1.3
Packages: +7
+++++++
Resolving: total 7, reused 7, downloaded 0, done

dependencies:
+ chalk 1.1.3 (2.4.1 is available)

PS D:\Develop\pnpmtests> pnpm update chalk
Already up-to-date

dependencies:
- chalk ^1.1.3

PS D:\Develop\pnpmtests>
```

### Expected behavior:
chalk@latest or chalk in semver range was installed

### Actual behavior:
chalk was removed from package.json (but was still there in shrinkwrap and node_modules)

### Additional information:

 - `node -v` prints: v8.11.3
 - Windows, OS X, or Linux?: Windows 10 (powershell)

**non major update works fine:**
```
PS D:\Develop\pnpmtests> pnpm install chalk@2.0.0
Packages: +7
+++++++
Resolving: total 7, reused 7, downloaded 0, done

dependencies:
+ chalk 2.0.0 (2.4.1 is available)

PS D:\Develop\pnpmtests> pnpm update chalk
Packages: +3 -3
+++---
Resolving: total 5, reused 5, downloaded 0, done

dependencies:
- chalk 2.0.0
+ chalk 2.4.1

PS D:\Develop\pnpmtests>
```

**Comment by jiripospisil** [comment]
Might be related to #1200

**Comment by zkochan** [comment]
@anantoghosh thanks for reporting. This is fixed in `v2.14.4`
